web/index.php

```diff
@@ -108,43 +105,47 @@
             $queryBuilder
                 ->andWhere('class_name = :class')
                 ->setParameter(':class', 'Outlandish\Wpackagist\Package\Theme');
+            $wporgQueryResults = $wporgClient->getThemesBy('search', $query, $currentPage, 30, ['slug']);
+            $slugs = array_column($wporgQueryResults['themes'], 'slug');
             break;
         case 'plugin':
             $queryBuilder
                 ->andWhere('class_name = :class')
                 ->setParameter(':class', 'Outlandish\Wpackagist\Package\Plugin');
+            $wporgQueryResults = $wporgClient->getPluginsBy('search', $query, $currentPage, 30, ['slug']);
+            $slugs = array_column($wporgQueryResults['plugins'], 'slug');
             break;
         default:
             break;
     }
 
-    switch ($active) {
-        case 1:
-            $queryBuilder->andWhere('is_active');
-            break;
-
-        default:
-            $queryBuilder->orderBy('is_active', 'DESC');
-            break;
-    }
-
     if (!empty($query)) {
+        // We match the results from the wordpress API to our database.
         $queryBuilder
-            ->andWhere('name LIKE :name')
-            ->addOrderBy('name LIKE :order', 'DESC')
-            ->addOrderBy('name', 'ASC')
-            ->setParameter(':name', "%{$query}%")
-            ->setParameter(':order', "{$query}%");
+            ->andWhere('p.name IN (:name)')
+            ->setParameter(':name', $slugs, \Doctrine\DBAL\Connection::PARAM_STR_ARRAY);
+
+        // This gymnastic is needed because the SQLITE database doesn't support ordering by FIELD, so we need to
+        // do it in PHP. We order the results from the database to match what the  wordpress API is returning us.
+        $statement = $queryBuilder->execute();
+        $packageResults = $statement->fetchAll(PDO::FETCH_ASSOC);
```

Comment:
I also get the following `Notice: Array to string conversion in /vagrant/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php on line 628`